src/main/java/gregtech/api/capability/impl/EnergyContainerBatteryBuffer.java

```diff
@@ -58,8 +58,9 @@ public long acceptEnergyFromNetwork(EnumFacing side, long voltage, long amperage
         return amperageUsed;
     }
 
-    private static boolean chargeItemWithVoltageExact(IElectricItem electricItem, long voltage, int tier, boolean simulate) {
-        return electricItem.charge(voltage, tier, false, simulate) == voltage;
+    private static boolean chargeItemWithVoltage(IElectricItem electricItem, long voltage, int tier, boolean simulate) {
+        long charged = electricItem.charge(voltage, tier, false, simulate);
+        return charged > 0 && charged <= voltage;
```

Comment:
Is that `&& charged <= voltage` really necessary?

Reply:
I am suggesting this change as there is no need to check other end as `charge` can't return it anyway
```suggestion
        return charged > 0;
```